syne_tune/blackbox_repository/blackbox_tabular.py

```diff
@@ -150,15 +150,44 @@ def hyperparameter_objectives_values(self):
         :return: X, y of shape (num_evals * num_seeds * num_fidelities, num_hps)
         and (num_evals * num_seeds * num_fidelities, num_objectives)
         """
+        objectives_evaluations = self.objectives_evaluations
+        hyperparameters = self.hyperparameters
+        if np.isnan(np.sum(objectives_evaluations)):
+            # Replace nan with previous value. Assumes that elapsed time is cumulative.
+            objectives_evaluations = objectives_evaluations.copy()
+            hyperparameters = hyperparameters.copy()
+            for config_idx in range(objectives_evaluations.shape[0]):
+                for seed_idx in range(objectives_evaluations.shape[1]):
+                    for fidelity_idx in range(objectives_evaluations.shape[2]):
+                        for objective_idx in range(objectives_evaluations.shape[3]):
+                            if np.isnan(
+                                objectives_evaluations[config_idx][seed_idx][
+                                    fidelity_idx
+                                ][objective_idx]
+                            ):
+                                objectives_evaluations[config_idx][seed_idx][
+                                    fidelity_idx
+                                ][objective_idx] = objectives_evaluations[config_idx][
+                                    seed_idx
+                                ][
+                                    fidelity_idx - 1
+                                ][
+                                    objective_idx
+                                ]
```

Comment:
Rather than calling shape 4 times, this would be more readable I think:

```suggestion
            num_configs, num_seeds, num_fidelities, num_objectives = objective.evaluations.shape
```